components/eventing-controller/pkg/backend/nats/jetstream/jetstream_integration_test.go

```diff
@@ -1398,3 +1418,124 @@ func TestJetStream_CheckConsumerConfig(t *testing.T) {
+	defer testEnvironment.jsClient.natsConn.Close()
+
+	testEnvironment.jsBackend.Config.JSStreamStorageType = StorageTypeFile
+	testEnvironment.jsBackend.Config.MaxReconnects = 0
+	initErr := jsBackend.Initialize(nil)
+	require.NoError(t, initErr)
+
+	subscriber := evtesting.NewSubscriber()
+	defer subscriber.Shutdown()
+	require.True(t, subscriber.IsRunning())
+
+	defaultMaxInflight := 9
+	defaultSubsConfig := env.DefaultSubscriptionConfig{MaxInFlightMessages: defaultMaxInflight}
+	sub := evtesting.NewSubscription("sub", "foo",
+		evtesting.WithNotCleanFilter(),
+		evtesting.WithSinkURL(subscriber.SinkURL),
+		evtesting.WithStatusConfig(defaultSubsConfig),
+	)
+	require.NoError(t, addJSCleanEventTypesToStatus(sub, testEnvironment.cleaner))
+
+	// when
+	err := jsBackend.SyncSubscription(sub)
+
+	// then
+	require.NoError(t, err)
+
+	// get cleaned subject
+	subject, err := backendnats.GetCleanSubject(sub.Spec.Filter.Filters[0], testEnvironment.cleaner)
+	require.NoError(t, err)
+	require.NotEmpty(t, subject)
+
+	require.Len(t, jsBackend.subscriptions, 1)
+	oldJsSubject := jsBackend.GetJetStreamSubject(subject)
+	oldJsSubKey := NewSubscriptionSubjectIdentifier(sub, oldJsSubject)
+	oldJsSub := jsBackend.subscriptions[oldJsSubKey]
+	require.NotNil(t, oldJsSub)
+	require.True(t, oldJsSub.IsValid())
+
+	// test if subscription is working properly by sending an event
+	// and checking if it is received by the subscriber
+	data := fmt.Sprintf("data-%s", time.Now().Format(time.RFC850))
+	expectedDataInStore := fmt.Sprintf("\"%s\"", data)
+	require.NoError(t, SendEventToJetStream(jsBackend, data))
+	require.NoError(t, subscriber.CheckEvent(expectedDataInStore))
+
+	// set metadata on NATS subscriptions
+	// so that we can later verify if the nats subscriptions are the same (not re-created by Sync)
+	msgLimit, bytesLimit := 2048, 2048
+	require.Len(t, jsBackend.subscriptions, 1)
+	for _, jsSub := range jsBackend.subscriptions {
+		require.True(t, jsSub.IsValid())
+		require.NoError(t, jsSub.SetPendingLimits(msgLimit, bytesLimit))
+	}
+
+	//shutdown the JetStream and start so that existing subscription gets invalid.
+	testEnvironment.natsServer.Shutdown()
+	require.Eventually(t, func() bool {
+		return !testEnvironment.jsBackend.conn.IsConnected()
+	}, 30*time.Second, 2*time.Second)
+
+	// when
+	// restart the NATS server
+	_ = evtesting.RunNatsServerOnPort(
+		evtesting.WithPort(testEnvironment.natsPort),
+		evtesting.WithJetStreamEnabled())
+	// the unacknowledged message must still be present in the stream
+
+	require.Eventually(t, func() bool {
+		info, err := testEnvironment.jsClient.StreamInfo(defaultStreamName)
+		require.NoError(t, err)
+		return info != nil && err == nil
+	}, 60*time.Second, 5*time.Second)
+
+	// SyncSubscription binds the existing subscription to JetStream created one
+	err = jsBackend.SyncSubscription(sub)
+	// then
+	require.NoError(t, err)
+
+	// given
+	// Now, change the filter in subscription
```

Comment:
(nit) why dont we delete the kyma subscription instead of changing the filter?

Reply:
yes we could also delete it. I just used change as I was testing with change. I think we can leave it so.